### PR TITLE
feat(dataarts): add resource dataarts architecture code table

### PR DESCRIPTION
**Pull request by profoundwu**
**What this PR does / why we need it**:
add resource dataarts architecture code table

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/dataarts" TESTARGS="-run TestAccArchitectureCodeTable_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dataarts -v -run TestAccArchitectureCodeTable_basic -timeout 360m -parallel 4
=== RUN   TestAccArchitectureCodeTable_basic
=== PAUSE TestAccArchitectureCodeTable_basic
=== CONT  TestAccArchitectureCodeTable_basic
--- PASS: TestAccArchitectureCodeTable_basic (36.45s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dataarts  36.490s

```

### CODE REVIEW

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go [hunk 1]
```diff
@@ -0,0 +1,758 @@
...
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The time when the code table was updated.",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the code table.",
+			},
+			"id": {
```
Comment (jinyangyang222) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go:150]: id is a reserved field and does not need to be declared in scehma

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go [hunk 2]
```diff
@@ -0,0 +1,758 @@
...
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "The ordinal of a field.",
+						},
+						"table_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The ID of the code table.",
+						},
+						"id": {
```
Comment (jinyangyang222) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go:111]: change to field_id

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go [hunk 3]
```diff
@@ -0,0 +1,758 @@
...
+			StateContext: resourceCodeTableImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The region where the user group is located.",
```
Comment (jinyangyang222) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go:47]: Please keep consistent with the document

File: docs/resources/dataarts_architecture_code_table.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+The `fields` block supports:
+
+* `name` - (Required, String) Specifies the name of a field.
+
+* `code` - (Required, String) Specifies the code of a field.
+
+* `type` - (Required, String) Specifies the type of a field. Valid values are: **BIGINT**, **BOOLEAN**, **DATE**,
+  **DECIMAL**, **DOUBLE**, **STRING**, and **TIMESTAMP**.
+
+* `values` - (Optional, List) Specifies the values of a field.
```
[the remark this answers is not in the document]
Comment (profoundwu) [reply docs/resources/dataarts_architecture_code_table.md]: fixed

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go (hunk 1 above)
Comment (profoundwu) [reply huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go]: fixed

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go (hunk 3 above)
Comment (profoundwu) [reply huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go]: fixed

File: huaweicloud/services/acceptance/acceptance.go [hunk 5]
```diff
@@ -1021,6 +1022,13 @@ func TestAccPreCheckDataArtsCdmName(t *testing.T) {
 	}
 }
 
+// lintignore:AT003
+func TestAccPreCheckDataArtsCodeTableDirectoryID(t *testing.T) {
+	if HW_DATAARTS_CODE_TABLE_DIRECTORY_ID == "" {
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/acceptance.go]: can we use `huaweicloud_dataarts_architecture_directory` to create the directory?

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go [hunk 6]
```diff
@@ -0,0 +1,753 @@
...
+	fieldsIDs, err := jmespath.Search("code_table_fields[*].id", codeTable)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving DataArts Architecture code table fields: %s is not found in API response", "id")
+	}
+
+	return fieldsIDs.([]interface{}), nil
+}
+
+func hasFieldValues(d *schema.ResourceData) bool {
+	count := false
```
Comment (ShiChangkuo) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go]: `has` or `ok` will make more sense.
```suggestion
	has := false
```

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go [hunk 7]
```diff
@@ -0,0 +1,753 @@
...
+
+	id, err := jmespath.Search("data.value.id", createCodeTableRespBody)
+	if err != nil || id == nil {
+		return diag.Errorf("error creating DataArts Architecture code table: %s is not found in API response", "id")
+	}
+
+	d.SetId(id.(string))
+
+	yes := hasFieldValues(d)
+	if yes {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go]: this is more golang style
```suggestion
	if yes := hasFieldValues(d); yes {
```

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go [hunk 8]
```diff
@@ -0,0 +1,753 @@
...
+	rawFields := d.Get("fields").([]interface{})
+	fields := make([]map[string]interface{}, len(rawFields))
+	for i, rawField := range rawFields {
+		fieldMap := rawField.(map[string]interface{})
+		field := make(map[string]interface{})
+		field["ordinal"] = i + 1
+		field["name_ch"] = fieldMap["name"].(string)
+		field["name_en"] = fieldMap["code"].(string)
+		field["data_type"] = fieldMap["type"].(string)
+		field["description"] = utils.ValueIngoreEmpty(fieldMap["description"].(string))
```
Comment (ShiChangkuo) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go]: this is more golang style
```suggestion
		field := map[string]interface{}{
			"ordinal": i + 1,
			"name_ch": fieldMap["name"].(string),
			"name_en": fieldMap["code"].(string),
			"data_type": fieldMap["type"].(string),
			"description": utils.ValueIngoreEmpty(fieldMap["description"].(string)),
		}
```

File: huaweicloud/services/acceptance/acceptance.go (hunk 5 above)
Comment (profoundwu) [reply huaweicloud/services/acceptance/acceptance.go]: fixed

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go (hunk 8 above)
Comment (profoundwu) [reply huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go]: fixed

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go [hunk 9]
```diff
@@ -0,0 +1,452 @@
...
+	return common.CheckDeletedDiag(d, parseCodeTableError(err), "error deleting DataArts Architecture code table")
+}
+
+func buildCodeTableDeleteBodyParams(d *schema.ResourceData) map[string]interface{} {
+	return map[string]interface{}{
+		"ids": []string{d.Id()},
+	}
+}
+
+func readCodeTable(client *golangsdk.ServiceClient, d *schema.ResourceData) (*http.Response, error) {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go:378]:
```suggestion
func readCodeTable(client *golangsdk.ServiceClient, id string) (*http.Response, error) {
```

**Pull request merged**